### PR TITLE
Merge ImPlot implementations

### PR DESCRIPTION
**Pull request by johannaheuser**

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Pulsed Power Monitoring
```
Comment (krystynast) [inline src/README.md:1]: OpenCMW - How to run
add 
- run ./opencmw_worker/build/InferenceTool 
- information of data files from the integrators

File: src/implot_visualization/app/deserializer/deserialize_json.h [hunk 2]
```diff
@@ -49,52 +59,111 @@ class IAcquisition {
...
+    void   deserialize();
+    void   fail();
+    double sumOfUsage();
+
+private:
+    void setSumOfUsageDay();
+    void setSumOfUsageWeek();
+    void setSumOfUsageMonth();
+};
+
```
Comment (krystynast) [inline src/implot_visualization/app/deserializer/deserialize_json.h:140]: change Inheitance to public

File: src/implot_visualization/app/deserializer/deserialize_json.h [hunk 3]
```diff
@@ -49,52 +59,111 @@ class IAcquisition {
...
+class RealPowerUsage : IAcquisition<PowerBuffer> {
+public:
+    // std::vector<std::string> signalNames;
+
+    uint64_t                 lastRefTrigger = 0;
+    std::string              jsonString     = "";
+    uint64_t                 lastTimeStamp  = 0.0;
+
+    double                   deliveryTime;
+
```
Comment (krystynast) [inline src/implot_visualization/app/deserializer/deserialize_json.h:150]: remove field success

File: src/implot_visualization/app/deserializer/deserialize_json.h [hunk 4]
```diff
@@ -49,52 +59,111 @@ class IAcquisition {
...
+    bool                     success            = false;
+    bool                     init               = false;
+
+    double                   realPowerUsageOrig = 0.0;
+    double                   realPowerUsage     = 0.0;
+
```
Comment (krystynast) [inline src/implot_visualization/app/deserializer/deserialize_json.h:156]: remove field buffers

File: src/implot_visualization/app/plot_tools/plot_tools.h [hunk 5]
```diff
@@ -197,7 +199,205 @@ void plotPower(std::vector<ScrollingBuffer> &signals, DataInterval Interval) {
...
+        ImGui::TableSetupColumn("Device", ImGuiTableColumnFlags_WidthFixed, 200.0f);
+        ImGui::TableSetupColumn("On/Off", ImGuiTableColumnFlags_WidthFixed, 50.0f);
+        ImGui::TableSetupColumn("Apparent Power [VA]", ImGuiTableColumnFlags_WidthFixed, 150.0f);
+        ImGui::TableSetupColumn(timePeriod.c_str());
+        ImGui::TableHeadersRow();
+
+        std::string output_simbol;
+        for (int row = offset; row < offset + len; row++) {
+            ImGui::TableNextRow();
+            ImGui::TableSetColumnIndex(0);
```
Comment (krystynast) [inline src/implot_visualization/app/plot_tools/plot_tools.h:298]: move  line 299 after 301

File: src/implot_visualization/app/plot_tools/plot_tools.h [hunk 6]
```diff
@@ -197,7 +199,205 @@ void plotPower(std::vector<ScrollingBuffer> &signals, DataInterval Interval) {
...
+            ImGui::Text("%s", powerUsage.devices[row].c_str());
+
+            if (powerUsage.init && row < powerUsage.devices.size()) {
+                ImGui::TableSetColumnIndex(1);
+                ImVec2 pos = window->DC.CursorPos;
+
+                // columns only for initialized values
```
Comment (krystynast) [inline src/implot_visualization/app/plot_tools/plot_tools.h:305]: Possible Issue ??:
powerUsage.powerUsages[row] - check size first

File: src/implot_visualization/app/plot_tools/plot_tools.h [hunk 7]
```diff
@@ -197,7 +199,205 @@ void plotPower(std::vector<ScrollingBuffer> &signals, DataInterval Interval) {
...
+                    break;
+                }
+            }
+        }
+        ImGui::EndTable();
+    }
+}
+
+void plotTable(PowerUsage &powerUsage, int m_d_w, double month_value = 0, double week_value = 0, double day_value = 0) {
+    printf("Ploting Table, init %d\n", powerUsage.init);
```
Comment (krystynast) [inline src/implot_visualization/app/plot_tools/plot_tools.h:354]: remove printf

File: src/implot_visualization/app/deserializer/deserialize_json.cpp [hunk 8]
```diff
@@ -162,15 +174,158 @@ void AcquisitionSpectra::deserialize() {
...
+    this->lastRefTrigger = refTrigger_ns;
+    addToBuffers(channelFrequencyValues, channelMagnitudeValues);
+}
+
+PowerUsage::PowerUsage() {
+}
+
+PowerUsage::PowerUsage(int _numSignals) {
+    std::vector<Buffer> _buffers(_numSignals);
+    this->buffers = _buffers;
```
Comment (krystynast) [inline src/implot_visualization/app/deserializer/deserialize_json.cpp:192]: remove code

File: src/implot_visualization/app/deserializer/deserialize_json.h [hunk 9]
```diff
@@ -49,52 +59,111 @@ class IAcquisition {
...
+    uint64_t                 lastRefTrigger = 0;
+    std::string              jsonString     = "";
+    std::vector<Buffer>      buffers;
+    bool                     success = false;
+    bool                     init    = false;
+    double                   deliveryTime;
+    int64_t                  timestamp;
+    uint64_t                 lastTimeStamp   = 0.0;
+
+    double                   powerUsageToday = 0.0;
```
Comment (johannaheuser) [inline src/implot_visualization/app/deserializer/deserialize_json.h:115]: Does powerUsage have a different meaning than kWhUsed?

**Pull request merged**